packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1696,6 +1696,60 @@ abstract class ParentDataWidget<T extends ParentData> extends ProxyWidget {
 /// for that inherited widget using [BuildContext.dependOnInheritedWidgetOfExactType]
 /// and then returns the [ThemeData].
 ///
+/// When using the `of` method, the `context` must be a descendant of the
+/// [InheritedWidget], meaning it must be "bellow" the [InheritedWidget] in the
+/// tree.
+///
+/// {@tool snippet}
+///
+/// DO
+///
+/// ```dart
+/// class MyPage extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return Scaffold(
+///       body: FrogColor(
+///         color: Colors.green,
+///         child: Builder(
+///           builder: (builderContext) {
+///             return Text(
+///               'Hello Frog',
+///               // note how `builderContext` is bellow `FrogColor`
+///               style: TextStyle(color: FrogColor.of(builderContext).color),
+///             );
+///           },
+///         ),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool snippet}
+///
+/// DON'T
```

Comment:
In this example, the context used is the one from the `MyOtherPage` widget, which is an ancestor of the `FrogColor` widget, so this does not work:

Reply:
done

Reply:
oh i also meant to remove "DON'T" (and "DO" above). The writing style for the API docs try to avoid telling people what to do, they just give information and let the reader decide what to do with it.

Reply:
removed